### DISCUSSION TITLE
AM2301B - I2C

### DISCUSSION THREAD

**Discussion by clauderobi** [arendst/Tasmota#14713] [Q&A] [answered]
The B variant of the AM2301 (so AM2301B) is I2C based. It is supported? If not is it planned?

**Comment by sfromis** [comment, marked as answer]
Such things are not "planned", but normally a question of someone doing the work and submitting a PR to the open source project. As both AM2320 and DHT12 are supported on I2C, it may not be a lot of work, if it is not already "compatible enough" to work.

**Reply by barbudor** [reply]
AM2301B is advertized to use addess 0x38
A very brief look at the datasheet of AHT10/AHT20 that are using the same address and from the same manufacturer seems to pushed the idea that it might compatible with those.
I would suggest you tests with those drivers. As those drivers are not included into any official build by default, you need either
- build your own custom firmware
- use one of the unofficial build at https://github.com/tasmota/install/tree/main/firmware/unofficial like tasmota-allsensors

It's then a matter of using I2CDriver command to enable just the one driver that you want to test (auto detection on I2C is not something reliable and should be avoided)

**Reply by clauderobi** [reply]
Yeap, I can confirm this work perfectly. With 1 extra step: somewhat the VEML6070 UV detector was active; I had to disable that one before enabling the AHT20 support worked.

Thanks.

**Reply by barbudor** [reply]
It is a must to disable drivers that are not in use, especially when components can use the same address.
Auto detection of component is barely possible on I2C as the standard do not include any identification mecanism.

So did you tested both AHT10 and AHT20 drivers with only the AHT20 working ?
It would be good to know exactly so we can update the doc accordingly

**Reply by sfromis** [reply]
Some drivers are able to detect if the "right" sensor is found on an I2C address, but not something you can bank on. Most of the time it does not hurt to have drivers active with nothing on the addresses for those out-of-work drivers.

**Reply by clauderobi** [reply]
I am fairly new to tasmota and I must say that it is a little confusing. Obviously, like everyone I look at the GUI rather than RTFM.... So, I selected module 0 and configured GPIO4 and 5 for I2C. After that I issued the I2CDriver command selecting index 43, which according to the doc covers both the AHT1x and AHT2.x sensors. Not sure how to force one vs the other; I used the tasmota-allsensors build 

If you tell me how to force AHT1X then I can give it a try.

**Reply by barbudor** [reply]
My bad, you are right.
Despite being listed on 2 different lines in the doc, they are both I2cDriver43
I'll update the doc

Thanks

**Reply by barbudor** [reply]
Ok, looking at the code, it's the same driver but depending if you compile with USE_AHT1X or USE_AHT2X it enabled one or the other.
Then the display on the Web GUI will be adapted
As this requires recompiling to change from one or the other, I won't both you with that.
Could you just please confirm which firmware you used ? tasmota-allsensors ? or tasmota-mega ? or another one ?
And confirm that the Web page display as ATH2X ?

Thanks

**Reply by clauderobi** [reply]
I used tasmota-allsensors. Effectively, the home page shows AHT2X

**Reply by barbudor** [reply]
Thanks for the confirmation
Doc has been updated regarding compatibility https://tasmota.github.io/docs/AHT1x/

**Reply by clauderobi** [reply]
I suggest to add a reference on https://tasmota.github.io/docs/AM2301/ to tell people with the B variant to look at https://tasmota.github.io/docs/AHT1x/

**Reply by barbudor** [reply]
It's already there 😎
On the AM2301 page now you can read
![image](https://user-images.githubusercontent.com/2996042/152654900-d42ce158-f038-4d02-9667-e749c4706a01.png)
